gettsim/pre_processing/piecewise_functions.py

```diff
@@ -1,31 +1,42 @@
 import numpy as np
 
 
-def piecewise_polynomial(
+def _piecewise_polynomial(
     x,
     lower_thresholds,
     upper_thresholds,
     rates,
     intercepts_at_lower_thresholds,
     rates_modified=False,
 ):
-    """Calculate value of the piecewise function at x. The function is defined by the
-    other input arrays.
+    """Calculate value of the piecewise function at `x`.
 
-    Args:
-        x (float, >0): The value that the function is applied to.
-        lower_thresholds (1-d array): The lower thresholds of each interval.
-        upper_thresholds (1-d array): The upper thresholds each interval.
-        rates (n-d arry): The slope in each interval. Where n is the degree of the
-        polynomial function.
-        intercepts_at_lower_thresholds (1-d array): The intercepts at the lower
-        threshold of each interval.
-        rates_modified: Boolean variable indicating, that intercepts can't be used
-        anymore.
+    This function works only for floats. To make it accept arrays for `x`, the function
```

Comment:
This function works only for scalar `x`. The public version `piecewise_polynomial` accepts 1-d arrays for `x`.

(rest is implementation detail and does not belong here but below)

Reply:
ok.